app/services/attribute_asserter.rb

```diff
@@ -15,10 +15,11 @@ class AttributeAsserter
     :phone
   ].freeze
 
-  def initialize(user, service_provider, authn_request)
+  def initialize(user, service_provider, authn_request, decrypted_pii)
     self.user = user
     self.service_provider = service_provider
     self.authn_request = authn_request
+    @decrypted_pii = decrypted_pii
```

Comment:
Private accessor sounds good to me, I didn't see the difference between memoized instance variables vs otherwise